pkg/cloud/services/ec2/instances.go

```diff
@@ -1008,6 +1022,56 @@ func (s *Service) ModifyInstanceMetadataOptions(instanceID string, options *infr
 	return nil
 }
 
+// GetDHCPOptionSetDomainName returns the domain DNS name for the VPC from the DHCP Options.
+func (s *Service) GetDHCPOptionSetDomainName(ec2client ec2iface.EC2API, vpcID *string) *string {
+	fmt.Println("vpcID", vpcID)
+	log := s.scope.GetLogger()
+
+	if vpcID == nil {
+		log.Error(errors.New("vpcID is nil"), "vpcID is nil")
+		return nil
+	}
+
+	vpcInput := &ec2.DescribeVpcsInput{
+		VpcIds: []*string{vpcID},
+	}
+
+	vpcResult, err := ec2client.DescribeVpcs(vpcInput)
+	if err != nil {
+		log.Error(err, "failed to describe VPC", "vpcID", *vpcID)
+		return nil
+	}
+
+	dhcpInput := &ec2.DescribeDhcpOptionsInput{
+		DhcpOptionsIds: []*string{vpcResult.Vpcs[0].DhcpOptionsId},
+	}
+
+	log.Info("Fetching DHCP Options to get the domain-name", "dhcpOptionsID", *vpcResult.Vpcs[0].DhcpOptionsId)
+
+	dhcpResult, err := ec2client.DescribeDhcpOptions(dhcpInput)
+	if err != nil {
+		log.Error(err, "failed to describe DHCP Options Set", "DhcpOptionsSet", *dhcpResult)
+		return nil
+	}
+
+	for _, dhcpConfig := range dhcpResult.DhcpOptions[0].DhcpConfigurations {
+		if *dhcpConfig.Key == "domain-name" {
+			if len(dhcpConfig.Values) == 0 {
+				return nil
+			}
+			domainName := dhcpConfig.Values[0].Value
+			// default domainName is 'ec2.internal' in us-east-1 and 'region.compute.internal' in the other regions.
+			if strings.Contains(*domainName, "ec2.internal") || strings.Contains(*domainName, "compute.internal") {
+				return nil
+			}
```

Comment:
This check wouldn't be sufficient if someone supplied a `domain-name` of `us-east-1.compute.internal` right?

Reply:
Yep, you're right.

Reply:
Yep, you're right.